packages/server-wallet/src/protocols/application.ts

```diff
@@ -28,6 +29,31 @@ const isFunded = ({app: {funding, supported}}: ProtocolState): boolean => {
   return funded;
 };
 
+const myTurnToFund = ({app}: ProtocolState): boolean => {
+  if (!app.supported) return false;
+  if (app.chainServiceRequests.indexOf('fund') > -1) return false;
+
+  const myDestination = app.participants[app.myIndex].destination;
+  const allocation = checkThat(app.supported?.outcome, isSimpleAllocation);
+  const currentFunding = app.funding(allocation.assetHolderAddress);
+  const allocationsBeforeMe = _.takeWhile(
+    allocation.allocationItems,
+    a => a.destination !== myDestination
+  );
+  const targetFunding = allocationsBeforeMe.map(a => a.amount).reduce(BN.add, BN.from(0));
```

Comment:
Should we make it clear that this assumes these facts?

1. each participant cares about a unique destination
2. each destination appears at most once

One reason to drop (2), for instance, is to support ledger top-ups with as few state updates as possible.

Reply:
👍 Is your thought to alter the logic to potentially drop these assumptions? Or to leave a comment stating the assumptions?

Reply:
I think a comment would do for now, since partial top ups are not soon on the roadmap